src/documentation/io/built-in-google-bigquery.md

```diff
@@ -0,0 +1,761 @@
+
+In addition, you can also write your own types that have a mapping function to
+`TableRow`, and you can use side inputs in all `DynamicDestinations` methods.
+
+<!-- Java specific -->
+
+{:.language-java}
+To use dynamic destinations, you must create a `DynamicDestinations` object and
+implement the following methods:
+
+{:.language-java}
+* `getDestination`: Returns an object that `getTable` and `getSchema` can use as
+  the destination key to compute the destination table and/or schema.
+
+* `getTable`: Returns the table (as a `TableDestination` object) for the
+  destination key.
+
+* `getSchema`: Returns the table schema (as a `TableSchema` object) for the
+  destination key.
+
+{:.language-java}
+Then, use `write().to` with your `DynamicDestinations` object. In this example,
+you have a `PCollection` of events, and each event contains a user ID field.
+Beam writes each user ID’s events to a separate table, with a separate schema
+per user ID.
+
+This example writes weather data into a different table for every year.
+
+```java
+{% github_sample /apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryWriteDynamicDestinations
+%}```
+```py
+# The Beam SDK for Python does not currently support dynamic destinations.
+```
+
+### Using time partitioning
+
+{:.language-py}
+> The Beam SDK for Python does not currently support time partitioning.
+
+BigQuery time partitioning divides your table into smaller partitions, which is
+called a [partitioned table](https://cloud.google.com/bigquery/docs/partitioned-tables).
+Partitioned tables make it easier for you to manage and query your data.
+
+<!-- Java specific -->
+
+{:.language-java}
+To use BigQuery time partitioning, use one of these two methods:
+
+{:.language-java}
+* `withTimePartitioning`: This method takes a `TimePartitioning` class, and is
+  only usable if you are writing to a single table.
+
+* `withJsonTimePartitioning`: This method is the same as
+  `withTimePartitioning`, but takes a JSON-serialized String object.
+
+{:.language-java}
+This example generates one partition per day.
+
+```java
+{% github_sample /apache/beam/blob/master/examples/java/src/main/java/org/apache/beam/examples/snippets/Snippets.java tag:BigQueryTimePartitioning
+%}```
+```py
+# The Beam SDK for Python does not currently support time partitioning.
+```
+
+
+## Limitations
+
+BigQueryIO currently has the following limitations.
+
+1. You can’t sequence the completion of a BigQuery write with other steps of
+   your pipeline.
+
+2. There is no efficient way to import a large time-partitioned table. As a
+   workaround, you can use dynamic destinations with a table decorator per
+   partition. However, this can create a lot of load jobs and you may run out
+   of quota (1000 jobs/day).
+
+3. If you are using the Beam SDK for Python, you may have import size quota
+   issues if you write a very large data set. As a workaround, you can partition
```

Comment:
partition the dataset (for example using Beam's Partition) transform and write to multiple BigQuery tables.